temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/WorkersAutoConfiguration.java

```diff
@@ -0,0 +1,244 @@
+import io.temporal.testing.TestWorkflowEnvironment;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionValidationException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.*;
+import org.springframework.context.event.ContextRefreshedEvent;
+
+/** Provides a client based on `spring.temporal.workers` section */
+@Configuration
+@EnableConfigurationProperties(TemporalProperties.class)
+@ConditionalOnClass(name = "io.temporal.worker.WorkerFactory")
+@Conditional(WorkersPresentCondition.class)
+@Import({WorkersAutoConfiguration.ActivityExplicitConfigProcessor.class})
+public class WorkersAutoConfiguration {
+  private static final Logger log = LoggerFactory.getLogger(WorkersAutoConfiguration.class);
+
+  private final TemporalProperties properties;
+  private final ConfigurableListableBeanFactory beanFactory;
+
+  public WorkersAutoConfiguration(
+      TemporalProperties properties, ConfigurableListableBeanFactory beanFactory) {
+    this.properties = properties;
+    this.beanFactory = beanFactory;
+  }
+
+  @Bean(name = "temporalWorkerFactory", destroyMethod = "shutdown")
+  public WorkerFactory workerFactory(
+      WorkflowClient workflowClient,
+      @Qualifier("temporalTestWorkflowEnvironment") @Autowired(required = false) @Nullable
+          TestWorkflowEnvironment testWorkflowEnvironment) {
+    if (testWorkflowEnvironment != null) {
+      return testWorkflowEnvironment.getWorkerFactory();
+    } else {
+      return WorkerFactory.newInstance(workflowClient);
+    }
+  }
+
+  @Bean(name = "temporalWorkers")
+  public Collection<Worker> workers(
+      WorkerFactory workerFactory,
+      @Qualifier("autoDiscoveredWorkflowImplementations") @Autowired(required = false) @Nullable
+          Collection<Class<?>> autoDiscoveredWorkflowImplementationClasses) {
+    Set<Worker> workers =
+        properties.getWorkers().stream()
+            .map(workerProperties -> newWorker(workerFactory, workerProperties))
+            .collect(Collectors.toSet());
```

Comment:
Per https://github.com/temporalio/sdk-java/pull/1305#discussion_r921230729, I just think that it should be an error to configure two workers for the same task queue (and otherwise reuse silently).

Reply:
If you are talking about verifying integrity of the user configuration - yeah, it makes sense.